src/models/UserMapper.php

```diff
@@ -385,4 +385,80 @@ protected function markUserVerified($user_id) {
         $verify_stmt->execute($verify_data);
         return true;
     }
+
+    /**
+     * Does the currently-authenticated user have right to edit this user?
+     *
+     * User must be the edited user, or a site admin
+     *
+     * @param int $user_id The identifier for the user to edit
+     * @return bool True if the user has privileges, false otherwise
+     */
+    public function thisUserHasAdminOn($user_id) {
+        // do we even have an authenticated user?
+        $loggedInUser = $this->_request->getUserId();
+        if(false !== $loggedInUser) {
+            // are we asking for access to the current user?
+            if($loggedInUser == $user_id) {
+                // user can edit themselves
+                return true;
+            }
+
+            // is this a site admin?
+            if($this->isSiteAdmin($loggedInUser)) {
+                return true;
+            }
+        } 
+        return false;
+    }
+
+    /**
+     * Update an existing user record
+     *
+     * @param array $user   An array of fields to change
+     * @param int   $userId The user to update
+     * @return bool True if successful
+     */
+    public function editUser($user, $userId) {
+        // Sanity check: ensure all mandatory fields are present.
+        $mandatory_fields = array(
+            'full_name',
+            'email',
+        );
+        $contains_mandatory_fields = !array_diff($mandatory_fields, array_keys($user));
+        if (!$contains_mandatory_fields) {
+            throw new Exception("Missing mandatory fields");
+        }
+
+        // encode the password
+        $user['password'] = password_hash(md5($user['password']), PASSWORD_DEFAULT);
```

Comment:
password is not mandatory and may not have been sent in, so please check that it exists before using.